packages/x-date-pickers/src/DigitalClock/DigitalClock.types.ts

```diff
@@ -0,0 +1,46 @@
+import { DigitalClockClasses } from './digitalClockClasses';
+import { UncapitalizeObjectKeys } from '../internals/utils/slots-migration';
+import { BaseClockProps, ExportedBaseClockProps } from '../internals/models/props/clock';
+
+export interface ExportedDigitalClockProps<TDate> extends ExportedBaseClockProps<TDate> {
+  /**
+   * Number representing the increment of precision at which to display minutes and seconds.
+   * @default 5
+   */
+  timeStep?: number;
+}
+
+export interface DigitalClockSlotsComponent {}
+
+export interface DigitalClockSlotsComponentsProps {}
+
+export interface DigitalClockProps<TDate>
+  extends ExportedDigitalClockProps<TDate>,
+    BaseClockProps<TDate> {
+  /**
+   * Override or extend the styles applied to the component.
+   */
+  classes?: Partial<DigitalClockClasses>;
+  /**
+   * Overrideable components.
+   * @default {}
+   * @deprecated Please use `slots`.
+   */
+  components?: DigitalClockSlotsComponent;
+  /**
+   * The props used for each component slot.
+   * @default {}
+   * @deprecated Please use `slotProps`.
+   */
+  componentsProps?: DigitalClockSlotsComponentsProps;
+  /**
+   * Overrideable component slots.
+   * @default {}
+   */
+  slots?: UncapitalizeObjectKeys<DigitalClockSlotsComponent>;
```

Comment:
Can you think of anything that can become a slot on these components? 🤔 
If not, maybe we should just remove these `components/slots`? 🤔

Reply:
I've added slots for a single list item, if not for overriding, then just for passing custom props. 🤔